packages/expo-bluetooth/ios/EXBluetooth/EXBluetooth+JSON.m

```diff
@@ -0,0 +1,290 @@
+  NSString *underlyingError = [[userInfo objectForKey:NSUnderlyingErrorKey] localizedDescription];
+  NSString *errorCode = [NSString stringWithFormat:@"%ld", (long) input.code];
+  return @{
+           @"code": errorCode,
+           @"description": EXNullIfEmpty(input.localizedDescription),
+           @"reason": EXNullIfEmpty(input.localizedFailureReason),
+           @"suggestion": EXNullIfEmpty(input.localizedRecoverySuggestion),
+           @"underlayingError": EXNullIfEmpty(underlyingError)
+           };
+}
+
++ (NSDictionary *)advertisementData_NativeToJSON:(NSDictionary<NSString *,id> *)input
+{
+  NSString *localNameKey = input[CBAdvertisementDataLocalNameKey];
+  NSNumber *txPowerLevel = input[CBAdvertisementDataTxPowerLevelKey];
+  NSNumber *isConnectable = input[CBAdvertisementDataIsConnectable];
+  
+  NSData *manufacturerData = input[CBAdvertisementDataManufacturerDataKey];
+  NSString *manufacturerDataValue = @"";
+  if (input[CBAdvertisementDataManufacturerDataKey] != nil) {
+    manufacturerDataValue = [NSString stringWithFormat:@"%lu bytes",(unsigned long)manufacturerData.length];
+  }
+
+  NSDictionary *serviceData = [[self class] serviceData_NativeToJSON:input[CBAdvertisementDataServiceDataKey]];
+  
+  NSArray *serviceUUIDs = [[self class] CBUUIDList_NativeToJSON:input[CBAdvertisementDataServiceUUIDsKey]];
+  NSArray *overflowServiceUUIDs = [[self class] CBUUIDList_NativeToJSON:input[CBAdvertisementDataOverflowServiceUUIDsKey]];
+  NSArray *solicitedServiceUUIDs = [[self class] CBUUIDList_NativeToJSON:input[CBAdvertisementDataSolicitedServiceUUIDsKey]];
+
+  return @{
+        @"localName": EXNullIfEmpty(localNameKey),
+        @"txPowerLevel": EXNullIfNil(txPowerLevel),
+        @"isConnectable": @([isConnectable boolValue]),
+        @"manufacturerData": EXNullIfEmpty(manufacturerDataValue),
+        @"serviceData": EXNullIfNil(serviceData),
+        @"serviceUUIDs": EXNullIfNil(serviceUUIDs),
+        @"overflowServiceUUIDs": EXNullIfNil(overflowServiceUUIDs),
+        @"solicitedServiceUUIDs": EXNullIfNil(solicitedServiceUUIDs)
+        };
+}
+
++ (NSString *)CBUUID_NativeToJSON:(CBUUID *)input
+{
+  if (!input) return nil;
+  // TODO: Bacon: Maybe we should return all of the data??
+  return [input UUIDString];
+}
+
+
++ (NSMutableArray<NSString *> *)CBUUIDList_NativeToJSON:(NSArray<CBUUID *> *)input
+{
+  if (!input) return nil;
+  
+  NSMutableArray *output = [NSMutableArray new];
+  for (CBUUID *value in input) {
+    NSString *serializedValue = [[self class] CBUUID_NativeToJSON:value];
+    [output addObject:EXNullIfEmpty(serializedValue)];
+  }
+  return output;
+}
+
++ (NSMutableArray<CBUUID *> *)CBUUIDList_JSONToNative:(NSArray *)input
+{
+  if (!input) return nil;
+  
+  NSMutableArray *output = [NSMutableArray new];
+  for (id value in input) {
+    if ([value isKindOfClass:[NSString class]]) {
+      // TODO: Bacon: Maybe add support for generating CBUUID from Data (ArrayBuffer)
```

Comment:
Probably should stick to strings, we're doing BLE I/O here, using strings vs. ArrayBuffers is not a meaningful performance comparison and in terms of an API IMO strings are more natural to work with